src/common/utils/collection-functions.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * Get the value behind `key`. If it was not pressent, first insert `value`
+ * @param map The map to interact with
+ * @param key The key to insert into the map with
+ * @param value The value to optional add to the map
+ * @returns The value in the map
+ */
+export function getOrInsert<K, V>(map: Map<K, V>, key: K, value: V): V {
+  if (map.has(key)) {
+    return map.get(key);
+  }
+
+  return map.set(key, value).get(key);
+}
+
+/**
+ * Get the value behind `key`. If it was not pressent, first insert the value returned by `getVal`
+ *
+ * This function should be used over `getInsertWith` if it is expensive to build a `V`
+ * @param map The map to interact with
+ * @param key The key to insert into the map with
+ * @param builder A function that returns a new instance of `V`.
+ * @returns The value in the map
+ */
+export function getOrInsertWith<K, V>(map: Map<K, V>, key: K, builder: () => V): V {
```

Comment:
This is not really used anywhere, why would we add "dead code"?

Reply:
I did a `git checkout` of this file from #3310 which is why the whole file was added. Will remove dead code.